src/openMVG/sfm/sfm_data_graph_utils.cpp

```diff
@@ -0,0 +1,139 @@
+// This file is part of OpenMVG, an Open Multiple View Geometry C++ library.
+
+// Copyright (c) 2015,2017 WhuAegeanSea, Pierre Moulon.
+
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "openMVG/sfm/sfm_data_graph_utils.hpp"
+#include "openMVG/sfm/pipelines/sfm_matches_provider.hpp"
+#include "openMVG/graph/connectedComponent.hpp"
+#include "openMVG/sfm/sfm_data.hpp"
+#include "openMVG/graph/graph.hpp"
+#include "openMVG/graph/graph_builder.hpp"
+#include "openMVG/features/feature.hpp"
+#include "openMVG/matching/indMatch.hpp"
+#include "openMVG/stl/stl.hpp"
+#include "openMVG/types.hpp"
+#include "openMVG/sfm/sfm_filters.hpp"
+
+#include <fstream>
+#include <iosfwd>
+
+namespace openMVG {
+namespace sfm {
+
+void KeepOnlyReferencedElement(
+	const std::set<IndexT> & set_remainingIds,
```

Comment:
Thx @whuaegeanse.
Please use 2spaces instead of tabs.
You can configure your code editor to do that. Or you can replace `\t` by two spaces thanks to search and replace.